pages/api/documents/[id]/index.ts

```diff
@@ -40,9 +41,53 @@ export default async function handle(
         error: (error as Error).message,
       });
     }
+  } else if (req.method === "DELETE") {
+    // DELETE /api/document/:id
+    const session = await getServerSession(req, res, authOptions);
+    if (!session) { 
+      res.status(401).end("Unauthorized");
+      return;
+    }
+
+    const { id } = req.query as { id: string };
+
+    try{
+      const document = await prisma.document.findUnique({
+        where: {
+          id: id,
+        },
+      });
+      
+      if (!document) {
+        return res.status(404).end("Document not found");
+      }
+      
+      // check that the user is owner of the document, otherwise return 401
+      if (document.ownerId !== (session.user as CustomUser).id) {
+        return res.status(401).end("Unauthorized to access the document");
+      }
+  
+      // delete the document from vercel blob 
+      await del(document.file);
+
+      // delete the document from database
+      await prisma.document.delete({
+        where: {
+          id: id,
+        }
+      })
+      
+      res.status(204).end();  // 204 No Content response for successful deletes
+    } catch (error) {
+      return res.status(500).json({
+        message: "Internal Server Error",
+        error: (error as Error).message,
+      });
+    }
+
   } else {
-    // We only allow GET and POST requests
-    res.setHeader("Allow", ["GET"]);
+    // We only allow GET, DELETE and POST requests
```

Comment:
```suggestion
    // We only allow GET and DELETE requests
```

I realized that there's no POST request in this API route